src/sparseml/optim/helpers.py

```diff
@@ -195,6 +195,68 @@ def parse_recipe_variables(
     return recipe_variables
 
 
+def _update_staged_container_value(var_key, var_value, container):
+    """
+    Extends the functionality of _update_container_value to handle staged containers.
+
+    :param var_key: The key we are attempting to find in the container.
+    :param var_value: The value which will overwrite the previous value
+        every time var_key is found in the container's keys.
+    :param container: A container generated from a YAML string of SparseML recipe
+    :return: (optionally mutated) container, as well as key_value
+        (True if var_key found in container's attributes, otherwise False)
+    """
```

Comment:
nit: pydocs only should be added for public methods, for private, can give inline comments about logic (ie any key at top level or within a stage of the recipe that matches `var_name` will be updated to `var_value`